### ISSUE TITLE
All letters uppercase on linux client

### ISSUE THREAD

**Issue by MNarath1** [input-leap/input-leap#2080]
### What happened?

When i connect between my windows server and linux client all my letters send from the windows server are always uppercase i cannot change that from the windows side only thing that changes stuff is if i use the capslock on my linux clients keyboard then it changes to lowercase with caps lock on but it cannot trigger capslock from the windows server nor can i ven use normal shift to upper or lower case letters.
Windows 10 (Input leap Server)
Fedora 41 KDE (Input leap Client) I am using Wayland here btw.

### Version

v3.0.2

### If applicable, where did you install Input Leap from?

On version 3.0.2

### What OSes are you seeing the problem on? (Check all that apply)

Linux

### If applicable, are you using Wayland or X11?

Wayland

### What OS versions are you using?

Fedora 41 KDE

**Comment by MNarath1** [comment]
the issue seesm to be gone now(after a reboot i think) with no real reason trough will leave it open in case someone else runs into it

**Comment by nbolton** [comment]
I wonder if your shift key was stuck on? That used to happen a lot in Deskflow/Synergy but we fixed it. I wonder if the fix was ported to Barrier/Input Leap?

**Comment by LuPro** [comment]
I've had this happen before with 3.0.2 and Linux Wayland Host -> Linux Wayland Client. I *think* it happens once you run and connect input leap once, then quit on the client machine and then restart/reconnect on the client - after that I get behavior as if caps lock state was switched until the next reboot (ie: all letters uppercase, but things like number keys and special symbols write the normal one). I've not had this happen on a Windows client so far.

I say I "think" because I'm not 100% sure this is what triggers it (running the client twice during one boot), but it definitely does stay until reboot once it occurs. I'm currently not able to test this, but possibly later today.

**Comment by pg3tech** [comment]
I had a similar problem. Wayland on both sides. Gnome on the server, KDE on the client...weirdly, the fix for me was to change the default "NumLock on Startup" setting to "off" under keyboard settings on the KDE side. Fixed it immediately.

Out of interest, are you using a TKL keyboard? I.e. a keyboard with no numpad? Just wondering if somehow that has something to do with it.

I have a TKL keyboard and it might be something to do with it.

**Comment by LuPro** [comment]
No, I have a full fat keyboard no TKL. I'll have to try if the Numlock thing fixes the issue for me, but that's at best a workaround since I do want to have my numpad on in general :sweat_smile:

**Comment by fpetry** [comment]
I also have this issue. Changing the "NumLock on Startup" setting, as @pg3tech suggested worked for a while. But the issue returned this morning.

I think it really is related to multiple starts of input leap.

Client and server both running on KDE/Wayland. Full keyboard.

**Comment by nbolton** [comment]
> I think it really is related to multiple starts of input leap.

Interesting. I wonder if accidentally running multiple instances also causes: 
- https://github.com/deskflow/deskflow/issues/7886
- https://github.com/deskflow/deskflow/issues/8005

**Comment by fpetry** [comment]
I just did a rather simple test:

* inputleap running on client and server, no CAPS Problem, everything fine
* on the client
  * press "stop" on tray menu
  * press "start" on tray menu
  * confirm Wayland share request; reconnect

--> ALL CAPS Problem appears when typing on client

Even after quitting (no processes in ps aux related to input-leap) and restarting input-leap, the problem stays.

**Comment by nbolton** [comment]
Please check to see if #2158 (merged) fixes this.

**Comment by fpetry** [comment]
I cannot test, as I cannot compile against libportal 0.8 under Ubuntu.

**Comment by virtuous-byte** [comment]
> Please check to see if [#2158](https://github.com/input-leap/input-leap/pull/2158) (merged) fixes this.

I was experiencing this on 3.0.2, built input-leap-git package in AUR, with commit e4c31b7 and it appears to be fixed.

**Comment by virtuous-byte** [comment]
@nbolton using Input Leap further I noticed another issue. Host environment is Windows and version 3.0.2-release-release-7e5889dc and on Linux it's from the AUR, 3.0.2-'git-2025-04-26-3b4a6c9f'-snapshot-

Problem is, on both systems I have Turkish Q layout. In this layout you write @ by using AltGr+Q combination, that's not working on Linux client right now. Any idea? Or does it need to be another GitHub issue?

I'd like to try with latest git on Windows but I don't know how to compile it on Windows.

**Comment by nbolton** [comment]
> e4c31b7

@sithlord48 Do you remember if this was fixed upstream?

**Comment by sithlord48** [comment]
No i do not
